external-crates/move/crates/move-model-2/src/summary.rs

```diff
@@ -688,8 +688,15 @@ impl Fields {
             N::StructFields::Native(_) => return,
         };
         self.positional_fields = is_positional;
+        let pos_name_of = |sym| {
+            if self.positional_fields {
+                Symbol::from(format!("pos{}", sym))
+            } else {
+                sym
+            }
+        };
```

Comment:
You might be able to do 
```suggestion
        let pos_name_of = 
           if self.positional_fields {
               |sym| Symbol::from(format!("pos{}", sym))
            } else {
               |sym| sym
            };
```
you might need to annotate `let pos_name_of: fn(Symbol) -> Symbol`

Reply:
Good suggestion -- Rust is totally happy with it and without any type annotations needed :)